src/core/kernel/events.cpp

```diff
@@ -48,7 +48,7 @@ bool Kernel::signalEvent(Handle handle) {
 		// We must reschedule our threads if we signalled one. Some games such as FE: Awakening rely on this
 		// If this does not happen, we can have phenomena such as a thread waiting up a higher priority thread,
 		// and the higher priority thread just never running
-		rescheduleThreads();
+		requireReschedule();
```

Comment:
I am not sure if this should happen.
Also need to investigate how OneShot events work a bit better